common/flatpak-dir.c

```diff
@@ -4065,6 +4065,9 @@ _flatpak_dir_ensure_repo (FlatpakDir   *self,
   if (self->repo != NULL)
     return TRUE;
 
+  if (g_cancellable_set_error_if_cancelled (cancellable, error))
+    return FALSE;
```

Comment:
If I understand correctly, the sprinkling of the cancellable checks is because this runs in a thread and the cancellable could have been cancelled at any moment. Maybe the first one should explicitly comment this.

Reply:
I will say my personal opinion is that each function that takes a GCancellable should have this check at the start of it. For example all of the OStree functions. But I understand the practicality of spreading it out here to make a change in one place.

Reply:
Correct. Furthermore, this is only a single place I faced the problem with. To be complete, one might spread the same checking all around the sources, but I did not dare to do it.